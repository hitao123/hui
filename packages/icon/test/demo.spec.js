import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Icon from '../index.vue';
import IconDemo from '../demo/index.vue';

describe('Icon component', () => {
  it('should show Icon', () => {
    const warpper = shallowMount(Icon, {
      propsData: {
        name: 'add'
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show IconDemo', () => {
    expect(renderToString(IconDemo)).toMatchSnapshot();
  });
});
