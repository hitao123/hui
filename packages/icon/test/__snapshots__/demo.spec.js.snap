// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo-icon">
  <div class="h-col h-col--8"><i class="h-icon h-icon-close" style="font-size:30px;"></i> <span>close</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-upgrade" style="font-size:30px;"></i> <span>upgrade</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-add-o" style="font-size:30px;"></i> <span>add-o</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-passed" style="font-size:30px;"></i> <span>passed</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-chat" style="font-size:30px;"></i> <span>chat</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-question" style="font-size:30px;"></i> <span>question</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-clock" style="font-size:30px;"></i> <span>clock</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-gold-coin" style="font-size:30px;"></i> <span>gold-coin</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-play" style="font-size:30px;"></i> <span>play</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-pause" style="font-size:30px;"></i> <span>pause</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-stop" style="font-size:30px;"></i> <span>stop</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-more-o" style="font-size:30px;"></i> <span>more-o</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-info-o" style="font-size:30px;"></i> <span>info-o</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-share" style="font-size:30px;"></i> <span>share</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-aim" style="font-size:30px;"></i> <span>aim</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-like-o" style="font-size:30px;"></i> <span>like-o</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-logistics" style="font-size:30px;"></i> <span>logistics</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-edit" style="font-size:30px;"></i> <span>edit</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-exchange" style="font-size:30px;"></i> <span>exchange</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-location" style="font-size:30px;"></i> <span>location</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-cart" style="font-size:30px;"></i> <span>cart</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-shop" style="font-size:30px;"></i> <span>shop</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-gift" style="font-size:30px;"></i> <span>gift</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-contact" style="font-size:30px;"></i> <span>contact</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-wap-home" style="font-size:30px;"></i> <span>wap-home</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-points" style="font-size:30px;"></i> <span>points</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-discount" style="font-size:30px;"></i> <span>discount</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-point-gift" style="font-size:30px;"></i> <span>point-gift</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-after-sale" style="font-size:30px;"></i> <span>after-sale</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-edit-data" style="font-size:30px;"></i> <span>edit-data</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-delete" style="font-size:30px;"></i> <span>delete</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-records" style="font-size:30px;"></i> <span>records</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-completed" style="font-size:30px;"></i> <span>completed</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-certificate" style="font-size:30px;"></i> <span>certificate</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-tosend" style="font-size:30px;"></i> <span>tosend</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-sign" style="font-size:30px;"></i> <span>sign</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-photo" style="font-size:30px;"></i> <span>photo</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-idcard" style="font-size:30px;"></i> <span>idcard</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-home" style="font-size:30px;"></i> <span>home</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-free-postage" style="font-size:30px;"></i> <span>free-postage</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-cash-back-record" style="font-size:30px;"></i> <span>cash-back-record</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-points-mall" style="font-size:30px;"></i> <span>points-mall</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-exchange-record" style="font-size:30px;"></i> <span>exchange-record</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-pending-payment" style="font-size:30px;"></i> <span>pending-payment</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-pending-orders" style="font-size:30px;"></i> <span>pending-orders</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-pending-deliver" style="font-size:30px;"></i> <span>pending-deliver</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-pending-evaluate" style="font-size:30px;"></i> <span>pending-evaluate</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-password-view" style="font-size:30px;"></i> <span>password-view</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-password-not-view" style="font-size:30px;"></i> <span>password-not-view</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-check" style="font-size:30px;"></i> <span>check</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-arrow" style="font-size:30px;"></i> <span>arrow</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-arrow-left" style="font-size:30px;"></i> <span>arrow-left</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-search" style="font-size:30px;"></i> <span>search</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-success" style="font-size:30px;"></i> <span>success</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-fail" style="font-size:30px;"></i> <span>fail</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-add" style="font-size:30px;"></i> <span>add</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-checked" style="font-size:30px;"></i> <span>checked</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-warn" style="font-size:30px;"></i> <span>warn</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-clear" style="font-size:30px;"></i> <span>clear</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-underway" style="font-size:30px;"></i> <span>underway</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-more" style="font-size:30px;"></i> <span>more</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-like" style="font-size:30px;"></i> <span>like</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-photograph" style="font-size:30px;"></i> <span>photograph</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-qr-invalid" style="font-size:30px;"></i> <span>qr-invalid</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-qr" style="font-size:30px;"></i> <span>qr</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-add2" style="font-size:30px;"></i> <span>add2</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-wechat" style="font-size:30px;"></i> <span>wechat</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-alipay" style="font-size:30px;"></i> <span>alipay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-wap-nav" style="font-size:30px;"></i> <span>wap-nav</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-ecard-pay" style="font-size:30px;"></i> <span>ecard-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-balance-pay" style="font-size:30px;"></i> <span>balance-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-peer-pay" style="font-size:30px;"></i> <span>peer-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-credit-pay" style="font-size:30px;"></i> <span>credit-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-debit-pay" style="font-size:30px;"></i> <span>debit-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-other-pay" style="font-size:30px;"></i> <span>other-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-shopping-cart" style="font-size:30px;"></i> <span>shopping-cart</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-browsing-history" style="font-size:30px;"></i> <span>browsing-history</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-goods-collect" style="font-size:30px;"></i> <span>goods-collect</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-shop-collect" style="font-size:30px;"></i> <span>shop-collect</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-receive-gift" style="font-size:30px;"></i> <span>receive-gift</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-send-gift" style="font-size:30px;"></i> <span>send-gift</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-setting" style="font-size:30px;"></i> <span>setting</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-coupon" style="font-size:30px;"></i> <span>coupon</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-gift-card-pay" style="font-size:30px;"></i> <span>gift-card-pay</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-cash-on-deliver" style="font-size:30px;"></i> <span>cash-on-deliver</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-phone" style="font-size:30px;"></i> <span>phone</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-description" style="font-size:30px;"></i> <span>description</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-card" style="font-size:30px;"></i> <span>card</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-value-card" style="font-size:30px;"></i> <span>value-card</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-gift-card" style="font-size:30px;"></i> <span>gift-card</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-hot" style="font-size:30px;"></i> <span>hot</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-new" style="font-size:30px;"></i> <span>new</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-new-arrival" style="font-size:30px;"></i> <span>new-arrival</span></div>
  <div class="h-col h-col--8"><i class="h-icon h-icon-hot-sale" style="font-size:30px;"></i> <span>hot-sale</span></div>
</div>
`;
